hadoop-common-project/hadoop-auth/src/test/java/org/apache/hadoop/security/authentication/util/TestRolloverSignerSecretProvider.java

```diff
@@ -52,7 +52,6 @@ public void testGetAndRollSecrets() throws Exception {
       Assert.assertEquals(2, allSecrets.length);
       Assert.assertArrayEquals(secret3, allSecrets[0]);
       Assert.assertArrayEquals(secret2, allSecrets[1]);
-      Thread.sleep(rolloverFrequency + 2000);
```

Comment:
As we are trying to improve this, let's do this test a little more controlled and use wairFor, etc.

Reply:
would you please explain more?

Reply:
I made a change to use a monitor object on the TRolloverSignerSecretProvider object, is this what are you looking for?